cli/npm/registry.rs

```diff
@@ -163,7 +186,19 @@ impl NpmRegistryApi {
         // attempt to load from the file cache
         maybe_package_info = self.load_file_cached_package_info(name);
       }
+
       if maybe_package_info.is_none() {
+        if self.cache_setting == CacheSetting::Only {
```

Comment:
This should probably be moved into `load_package_info_from_registry` in case someone calls that another way and forgets to check this.